src/System.Private.CoreLib/src/System/Lazy.cs

```diff
@@ -125,6 +125,19 @@ public Lazy(Func<T> valueFactory)
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="T:System.Threading.Lazy{T}"/> class that
+        /// uses a pre-initialized specified value.
+        /// </summary>
+        /// <remarks>
+        /// An instance created with this constructor should be usable by multiple threads
+        //  concurrently.
+        /// </remarks>
+        public Lazy(T value)
+        {
+            m_boxed = new Boxed(value);
```

Comment:
Ah done thanks. IMO all this shared code should be shared by having a submodule to something like core-commons repo.